### PR TITLE
Issue / 8620 PAX Update Conversion Tracking Service to Use Resolver in AdsModule Data Store

### PR DESCRIPTION
**Pull request by 10upsimon**
## Summary

Addresses issue:

- #8620

## Relevant technical choices

- Updated PAX services at `assets/js/modules/ads/pax/services.js`
    - Expanded logic for `conversionTrackingService.getSupportedConversionLabels()` to use new resolver in Ads `moduleData` store to retrieve supported conversion event labels
- Updated PAX services test suite at `assets/js/modules/ads/pax/services.test.js`
    - Included test to ensure we resolve to an empty array `[]` of no supported events are present
    - Included test to ensure we resolve to expected array of supported events when events are present via the `_googlesitekitModulesData` global var

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by 10upsimon** [comment]
@benbowler noting here that this PR was based off the same base branch for #8619 and therefore will include changes in that PR as well, the only applicable files to review here are:

- `assets/js/modules/ads/pax/services.js`
- `assets/js/modules/ads/pax/services.test.js`

Other changes should disappear once #8619 is merged. We can review so long and wait until that is merged for final approval.

**Review by benbowler — APPROVED** [review]
Hey @10upsimon approving the code review after testing and reading the code.

I have two process comments/suggestions:
1. Branch naming should use the type of the ticket so I believe this branch should have been named `enhancement/8620-pax-update...` rather than `issue/8620-pax-update...`.
2. Also, as this branch relies on changes in another branch, in future I suggest rebasing onto that branch and setting that branch as the base when creating the PR in GitHub:

Screenshot 2024-05-08 at 13 59 19

If you do this then only the new changes will be shown and GitHub is clever enough to change the base of the PR to develop once that branch is merged. I tried changing the base of this PR for you but the branches commits have diverged which mean's it not possible without a rebase on a local repo, which isn't worth it here but something for future.

**Review by tofumatt — APPROVED** [review]

**Pull request merged**
